.github/workflows/deploy.yml

```
@@ -0,0 +1,105 @@
+    - name: Deploy to Staging
+      run: |
+        echo "Deploying to staging environment..."
+        # Add your staging deployment commands here
+        # e.g., kubectl apply, ssh deploy, etc.
+
+  deploy-production:
+    runs-on: ubuntu-latest
+    needs: build-and-push
+    environment: production
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v')
+    
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)